app/scripts/metamask-controller.js

```diff
@@ -2388,6 +2389,30 @@ export default class MetamaskController extends EventEmitter {
     );
 
     ///: BEGIN:ONLY_INCLUDE_IF(snaps)
+    // Record Snap metadata whenever a Snap is added to state.
+    this.controllerMessenger.subscribe(
```

Comment:
This should be removed and upstreamed to: https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/snaps/SnapController.ts#L2574-L2580

Reply:
Done https://github.com/MetaMask/snaps/pull/2157